fc/parsing/actions.py

```diff
@@ -746,26 +805,50 @@ def resolve_namespaces(self, ns_map):
         for item in itertools.chain(self.inputs, self.outputs, self.optional_decls):
             item.ns_uri = ns_map[item.ns_prefix]
 
-    def associate_model(self, model, units):
-        """Tell this interface what model it is being used to manipulate.
+    def modify_model(self, model, units):
+        """Use the definitions in this interface to transform the provided model.
 
-        :param cellmlmanip.model.Model model: the model, for resolving variable references via ontology terms
+        :param cellmlmanip.model.Model model: the model to manipulate
         :param cellmlmanip.units.UnitStore units: the protocol's unit store, for resolving unit references
         """
-        self._model = model
-        self._units = units
+        self.model = model
```

Comment:
Do we need to store these here? The method description suggests this can be used on several models, so would prefer to pass model & units along to any method that needs them instead of storing

Reply:
I'm not sure we can easily pass them around, as pretty much everything needs them, including things like the symbol and number generators that are themselves passed around as functions. Since the `modify_model` method is treated as the main entry point, everything needing them will have them set correctly. But in any case all our uses at present only modify one model.